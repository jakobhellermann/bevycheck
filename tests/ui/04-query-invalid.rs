#![allow(unused)]
use bevy::prelude::*;

struct Foo;
struct Bar;

#[bevycheck::system]
fn system(_: Query<(Entity, Foo, Option<&Foo>, Flags<Foo>), Bar>) {}

fn main() {
    system.system();
}
