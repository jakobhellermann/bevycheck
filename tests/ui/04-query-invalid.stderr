error: invalid query parameter
 --> $DIR/04-query-invalid.rs:8:29
  |
8 | fn system(_: Query<(Entity, Foo, Option<&Foo>, Flags<Foo>), Bar>) {}
  |                             ^^^
  |
  = note: `Foo` is not a valid query type
  = help: if you want to query for a resource, use `&Foo` or `&mut Foo`

error: invalid query filter
 --> $DIR/04-query-invalid.rs:8:61
  |
8 | fn system(_: Query<(Entity, Foo, Option<&Foo>, Flags<Foo>), Bar>) {}
  |                                                             ^^^
  |
  = note: `Bar` is not a valid query filter
  = help: if you want to check for Bar's existence, use `With<Bar>``
